code/game/objects/effects/mines.dm

```diff
@@ -5,59 +5,95 @@
 	anchored = TRUE
 	icon = 'icons/obj/weapons/items_and_weapons.dmi'
 	icon_state = "uglymine"
+	base_icon_state = "uglymine"
 	/// We manually check to see if we've been triggered in case multiple atoms cross us in the time between the mine being triggered and it actually deleting, to avoid a race condition with multiple detonations
 	var/triggered = FALSE
 	/// Can be set to FALSE if we want a short 'coming online' delay, then set to TRUE. Can still be set off by damage
 	var/armed = TRUE
 	/// If set, we default armed to FALSE and set it to TRUE after this long from initializing
 	var/arm_delay
 
+	/// Who's got their foot on the mine's pressure plate
+	/// Stepping on the mine will set this to the first mob who stepped over it
+	/// The mine will not detonate via movement unless the first mob steps off of it
+	var/datum/weakref/foot_on_mine
+
 /obj/effect/mine/Initialize(mapload)
 	. = ..()
 	if(arm_delay)
 		armed = FALSE
-		icon_state = "uglymine-inactive"
+		update_appearance(UPDATE_ICON_STATE)
 		addtimer(CALLBACK(src, PROC_REF(now_armed)), arm_delay)
+
 	var/static/list/loc_connections = list(
 		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+		COMSIG_ATOM_EXITED = PROC_REF(on_exited),
 	)
 	AddElement(/datum/element/connect_loc, loc_connections)
 
 /obj/effect/mine/examine(mob/user)
 	. = ..()
 	if(!armed)
-		. += span_info("\tIt appears to be inactive...")
+		. += span_info("It appears to be inactive...")
+
+	var/atom/movable/unlucky_sod = foot_on_mine?.resolve()
+	if(user == unlucky_sod)
+		. += span_bolddanger("The pressure plate is depressed. Any movement you make will set it off now.")
+	else if(!isnull(unlucky_sod))
+		. += span_danger("The pressure plate is depressed by [unlucky_sod]. Any move they make'll set it off now.")
+
+/obj/effect/mine/update_icon_state()
+	. = ..()
+	if(armed)
+		icon_state = base_icon_state
+	else
+		icon_state = "[base_icon_state]-inactive"
 
 /// The effect of the mine
 /obj/effect/mine/proc/mineEffect(mob/victim)
-	to_chat(victim, span_danger("*click*"))
+	return
 
 /// If the landmine was previously inactive, this beeps and displays a message marking it active
 /obj/effect/mine/proc/now_armed()
 	armed = TRUE
-	icon_state = "uglymine"
+	update_appearance(UPDATE_ICON_STATE)
 	playsound(src, 'sound/machines/nuke/angry_beep.ogg', 40, FALSE, -2)
 	visible_message(span_danger("\The [src] beeps softly, indicating it is now active."), vision_distance = COMBAT_MESSAGE_RANGE)
 
-/obj/effect/mine/proc/on_entered(datum/source, atom/movable/AM)
+/obj/effect/mine/proc/can_trigger(atom/movable/on_who)
+	if(triggered || !isturf(loc) || iseffect(on_who) || !armed)
+		return FALSE
+	return TRUE
+
+/obj/effect/mine/proc/on_entered(datum/source, atom/movable/arrived)
 	SIGNAL_HANDLER
 
-	if(triggered || !isturf(loc) || !armed)
+	if(!can_trigger(arrived))
+		return
+	if(arrived.movement_type & FLYING)
 		return
 
-	if(AM.movement_type & FLYING)
+	foot_on_mine = WEAKREF(arrived)
```

Comment:
Doesn't this mean only the last person to step on the mine can't leave it? Like if person A steps on the mine, then person B, and then person B steps off it with person A still on it will detonate?

Reply:
Doesn’t this mean that if person A steps on a landmine, they wont be able to be tagged out by a person B? This seems like it just inverts the problem. As long as a person is ontop, it shouldn’t do kaboom (at least that’s what the PR desc seems to describe?)

Just check if there’s a valid other person standing on the landmine everytime someone steps of while it’s armed and set them as the foot_on_mine person for the fluff stuff